.github/workflows/java.yml

```
@@ -125,3 +125,29 @@ jobs:
       - name: Test
         shell: bash
         run: ci/scripts/java_test.sh $(pwd) $(pwd)/build
+
+  windows:
+    name: Windows Server 2022 Java JDK ${{ matrix.jdk }}
+    runs-on: windows-latest
+    if: ${{ !contains(github.event.pull_request.title, 'WIP') }}
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        jdk: [8]
+    steps:
+      - name: Set up Java
+        uses: actions/setup-java@v1
```

Can we use `@v3` here?

```suggestion
        uses: actions/setup-java@v3
```